### PR TITLE
Add possibility to override serializers

### LINKED ISSUE #8: Render embeds
I need to include YouTube videos that are embedded in a document. Since it would be hard to offer such a thing from this library (there are just too many ways to embed stuff), I would build a thing where you could override the `serializerWithStyle` with own renderers. Maybe by adding a dictionary that you can give to the app with all renderers you want to implement on your own. That way you also could implement a `h1` renderer if you prefer or the needed `embed` type renderer.

What do you think?

### PR DESCRIPTION
**Pull request by bitboxer**
A proposal to make it possible to add own ways how an element is displayed on the screen. Open for discussing how it finally will look like.

Closes #8

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -72,4 +99,18 @@ export type RichTextStyles = {
...
 }
 
+export interface SerializerFunction {
+  (
+    type: RichTextElementType,
+    element: SpanType,
+    text: string,
+    children: React.ComponentElement<TextProps, Text>,
+    index: string
+  ): React.ComponentElement<TextProps, Text> | null
```
Comment (Spoutnik97) [inline typings/index.d.ts]: if you type the return type as a Text, you cannot return a <Youtube /> component I guess.
@bitboxer what do you think about this?

Comment (bitboxer) [reply typings/index.d.ts]: Ah, yeah...this needs a broader skope then.

Comment (bitboxer) [reply typings/index.d.ts]: Short heads up: Will continue working on this Wednesday, Tuesday is a holiday here in Berlin and I took a vacation day on Monday.

Comment (bitboxer) [reply typings/index.d.ts]: @Spoutnik97 I am now using `JSX.Element`. Hope this is okay. It basically is a `React.Component` with an any type. Now everyone can return whatever they prefer.

**Comment by bitboxer** [comment]
@Spoutnik97 I updated this PR. No worries if you have more important things going on, just wanted to ping in case you didn't get notified about it.

File: typings/index.d.ts (hunk 1 above)
Comment (Spoutnik97) [reply typings/index.d.ts]: Perfect!

**Review by Spoutnik97 — APPROVED** [review]

**Comment by Spoutnik97** [comment]
Thanks @bitboxer, I missed the github notification :)

**Pull request merged**